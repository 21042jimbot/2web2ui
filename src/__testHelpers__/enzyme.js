export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
