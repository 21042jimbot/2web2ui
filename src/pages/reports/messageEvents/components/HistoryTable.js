import React, { Component } from 'react';

import { TableCollection } from 'src/components';
import { Panel } from '@sparkpost/matchbox';
import { Header, Row, TableWrapper } from './TableElements';

import _ from 'lodash';

class HistoryTable extends Component {
  static defaultProps = {
    messageHistory: []
  }

  createRows = () => {
    const { messageHistory, selectedId, handleEventClick } = this.props;

    return messageHistory.map((row) => {
      const selected = row.event_id === selectedId;
      return {
        ...row,
        selected,
        onClick: !selected ? () => handleEventClick(row.event_id) : null
      };
    });
  }

  render() {
    const rows = this.createRows();

    const disclaimer = _.get(rows[rows.length - 1], 'type') !== 'injection'
      ? <small>Previous events may be outside our 10 day storage range.</small>
      : null;

    return (
      <div>
        <Panel title='Message History' actions={[{ content: 'Refresh', onClick: this.props.handleRefresh }]}>
          <TableCollection
            outerWrapper={TableWrapper}
            headerComponent={Header}
            rowComponent={Row}
            rows={rows}
            pagination={false}
          />
        </Panel>
        { disclaimer }
      </div>
    );
  }
}

export default HistoryTable;
