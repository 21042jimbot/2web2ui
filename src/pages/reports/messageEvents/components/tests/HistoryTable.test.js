import React from 'react';
import { shallow } from 'enzyme';
import HistoryTable from '../HistoryTable';

describe('HistoryTable Component', () => {
  it('should render with no props', () => {
    const wrapper = shallow(<HistoryTable />);
    expect(wrapper).toMatchSnapshot();
  });

  const props = {
    handleEventClick: jest.fn(),
    messageHistory: [{ event_id: '1' }, { event_id: '2' }],
    selectedId: '2'
  };

  it('should render with all props', () => {
    const wrapper = shallow(<HistoryTable {...props} />);
    expect(wrapper.find('TableCollection')).toMatchSnapshot();
  });

  it('should not show disclaimer if first event is injection', () => {
    props.messageHistory.push({ event_id: '3', type: 'injection' });
    const wrapper = shallow(<HistoryTable {...props} />);
    expect(wrapper.find('small')).toHaveLength(0);
  });
});
