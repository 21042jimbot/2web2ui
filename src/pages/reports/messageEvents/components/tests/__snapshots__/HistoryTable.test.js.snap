// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryTable Component should render with all props 1`] = `
<TableCollection
  headerComponent={[Function]}
  outerWrapper={[Function]}
  pagination={false}
  rowComponent={[Function]}
  rows={
    Array [
      Object {
        "event_id": "1",
        "onClick": [Function],
        "selected": false,
      },
      Object {
        "event_id": "2",
        "onClick": null,
        "selected": true,
      },
    ]
  }
/>
`;

exports[`HistoryTable Component should render with no props 1`] = `
<div>
  <Panel
    actions={
      Array [
        Object {
          "content": "Refresh",
          "onClick": undefined,
        },
      ]
    }
    title="Message History"
  >
    <TableCollection
      headerComponent={[Function]}
      outerWrapper={[Function]}
      pagination={false}
      rowComponent={[Function]}
      rows={Array []}
    />
  </Panel>
  <small>
    Previous events may be outside our 10 day storage range.
  </small>
</div>
`;
